feat: LEAP-1097: Show Review buttons in Quick View

hlomzik opened the pull request:
If that's the task of another person we will show Accept/Reject. Part of "Reviewer Flow" experiment, hidden behind FF.

- [x] Check if user is of Manager role
- [x] Check if we are in Quick View

### PR fulfills these requirements
- [ ] Tests for the changes have been added/updated
- [ ] Docs have been added/updated
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance

### Describe the reason for change
_(if it's not clear from main description — otherwise remove this section)_

### What feature flags were used to cover this change?
`fflag_feat_all_leap_1081_reviewer_flow_updates`

### This change affects (describe how if yes)
- [ ] Performance
- [ ] Security
- [x] UX

### Does this PR introduce a breaking change?
- [x] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
- [ ] e2e (codecept)
- [ ] integration (cypress)
- [ ] unit (jest)

hlomzik committed:
feat: LEAP-1097: Show Review buttons in Quick View

If that's the task of another person we will show Accept/Reject.
Works only with FF on.

hlomzik committed:
Add FF for new Reviewer Flow

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9290887899

hlomzik committed:
Fix condition to display review buttons

Some hacks involved, but we don't have enough explicit info in LSF.

hlomzik committed:
Merge branch 'develop' into fb-leap-1097/review-in-quick-view

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9366454818

hlomzik committed:
Move "can be reviewed" logic to Annotation

to reuse it in hotkey handler

hlomzik committed:
Pass the whole SDK to lsfInit event

hlomzik committed:
merge

hlomzik committed:
Merge branch 'develop' into tmp/reviewer-flow

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9376697929

hlomzik commented on web/libs/editor/src/stores/Annotation/Annotation.js:
```diff
@@ -339,6 +340,23 @@ export const Annotation = types
+        && self.user?.email
+        && store.user?.email !== self.user?.email
+        // we have this only in LSE
+        && store.hasInterface("annotations:history")
```
most probably it's also added as part of default interfaces for LSO, so we should find another way to distinguish LSO and LSE. @Gondragos do you have any ideas?

Gondragos replied:
What about just defining a new interface for exactly that purpouse?

Gondragos replied:
Or.... there is an option
```js
getEnv(self).events.hasEvent("acceptAnnotation")
```

hlomzik committed:
Check for acceptAnnotation event to detect LSE

hlomzik committed:
Merge branch 'fb-leap-1097/review-in-quick-view' of github.com:heartexlabs/label-studio into fb-leap-1097/review-in-quick-view

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9398654449

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9418257022)
> Successfully merged:  7 files changed, 109 insertions(+), 150 deletions(-)

hlomzik committed:
Merge branch 'develop' into 'fb-leap-1097/review-in-quick-view'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9418257022

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9418283570

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9447685645)
> Successfully merged:  34 files changed, 1598 insertions(+), 78 deletions(-)

hlomzik committed:
Merge branch 'develop' into 'fb-leap-1097/review-in-quick-view'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9447685645

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9447706966

hlomzik committed:
Fix linting

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9449707001
